solr/solrj/src/java/org/apache/solr/common/cloud/ZkCredentialsInjector.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.common.cloud;
+
+import java.util.List;
+
+/**
+ * A class that retrieves Zookeeper credentials from some source to be injected into {@link
+ * DigestZkCredentialsProvider} and {@link DigestZkACLProvider}. The source here can be System
+ * Props, a file, a Secret Manager, or any other local or remote source.
+ */
+public interface ZkCredentialsInjector {
+
+  /**
+   * @return List of {@link ZkCredential}s representing Zookeeper credentials including the
+   *     username, the password and the permissions (ALL or READ)
+   */
+  List<ZkCredential> getZkCredentials();
+
+  class ZkCredential {
+
+    public enum Perms {
+      ALL,
+      READ
+    }
+
+    private String username;
+    private String password;
+    private String perms;
+
+    public ZkCredential() {}
+
+    public ZkCredential(String username, String password, Perms perms) {
+      this(username, password, String.valueOf(perms));
+    }
+
+    public ZkCredential(String username, String password, String perms) {
+      this.username = username;
+      this.password = password;
+      this.perms = perms;
+    }
+
+    public String getUsername() {
+      return username;
+    }
+
+    public String getPassword() {
```

Comment:
this is outside of the scope of the PR, but I really hate seeing passwords stored as string objects.

Reply:
Yes, I thought about it too, but wanted to keep this PR as simple as possible (in this first version)... besides, checked other places in Solr code and found out strings are used for storing passwords too, so I thought to do the same for now, and change it later on in a separate PR.
BTW, the existing code (_VMParamsAllAndReadonlyDigestZkACLProvider_/_VMParamsSingleSetCredentialsDigestZkCredentialsProvider_) is using a String to store the password too.